src/hip_module.cpp

```diff
@@ -652,3 +652,145 @@ hipError_t hipModuleGetTexRef(textureReference** texRef, hipModule_t hmod, const
     *texRef = reinterpret_cast<textureReference*>(addr);
     return ihipLogStatus(hipSuccess);
 }
+
+hipError_t ihipOccupancyMaxPotentialBlockSize(uint32_t* gridSize, uint32_t* blockSize,
+                                              hipFunction_t f, size_t dynSharedMemPerBlk,
+                                              uint32_t blockSizeLimit)
+{
+    using namespace hip_impl;
+
+    auto ctx = ihipGetTlsDefaultCtx();
+    hipError_t ret = hipSuccess;
+
+    if (ctx == nullptr) {
+        ret = hipErrorInvalidDevice;
+    }
+
+    hipDeviceProp_t prop{};
+    hipGetDeviceProperties(&prop, ihipGetTlsDefaultCtx()->getDevice()->_deviceId);
```

Comment:
Replace with ihipGetDeviceProperties() otherwise it will mess up the HIP trace.

Reply:
Fixed.